pkg/kv/dist_sender.go

```diff
@@ -1308,6 +1308,11 @@ func (ds *DistSender) sendToReplicas(
 			ds.metrics.SlowRequestsCount.Inc(1)
 			defer ds.metrics.SlowRequestsCount.Dec(1)
 
+		case <-ctx.Done():
+			// Caller has given up.
+			log.Eventf(ctx, "context finished during distsender send: %v", ctx.Err())
+			return nil, ctx.Err()
```

Comment:
I believe this needs to be treated as an ambiguous error if `haveCommit` is true.

Reply:
Why is that? It seems to me that if the context here has been cancelled, it always indicates that someone upstream cancelled the request or the query that emitted the request. In that case, it seems safe to just give up and return the raw context cancellation error.

That being said, there is some funniness with context cancellations in the system in general. A context cancellation on the server side of a gRPC stream seems to turn into an opaque `roachpb.internalError` at least some of the time. I'm not sure if that's intended.